src/EditorFeatures/Core/Implementation/SmartIndent/AbstractIndentationService.cs

```diff
@@ -46,7 +49,7 @@ private IEnumerable<IFormattingRule> GetFormattingRules(Document document, int p
             // the next line).  If we're in the latter case, we defer to the Formatting engine
             // as we need it to use all its rules to determine where the appropriate location is
             // for the following tokens to go.
-            if (ShouldUseSmartTokenFormatterInsteadOfIndenter(formattingRules, root, lineToBeIndented, documentOptions, cancellationToken))
+            if (!force && ShouldUseSmartTokenFormatterInsteadOfIndenter(formattingRules, root, lineToBeIndented, documentOptions, cancellationToken))
```

Comment:
this part of the change was necessary because we want to be able to compute the indentation without all the complex logic brought in by this Indenter/SmartTokenFormatter split.    I'd like a single entry-point that could give us that, but it's pretty complex to refactor this code to behave accordingly.  See https://github.com/dotnet/roslyn/issues/15711 for more details.